src/modules/tas_optimizer/mod.rs

```diff
@@ -333,15 +332,21 @@ fn optim_init(marker: MainThreadMarker, path: PathBuf, first_frame: usize) {
 
     // TODO: get current parameters.
     let parameters = unsafe {
-        let game_dir = CStr::from_ptr(engine::com_gamedir.get(marker).cast())
-            .to_str()
-            .unwrap();
-
-        let is_paranoia: bool = game_dir.eq("paranoia");
-        let is_cstrike: bool = game_dir.eq("cstrike");
-        let is_czero: bool = game_dir.eq("czero");
-
-        let pmove: *mut playermove_s = *engine::pmove.get(marker);
+        // Safety: the reference does not outlive this block, and com_gamedir can only be modified
+        // at engine start and while setting the HD models or the addon folder.
+        let game_dir = engine::com_gamedir
+            .get_opt(marker)
+            .and_then(|dir| CStr::from_ptr(dir.cast()).to_str().ok())
+            .unwrap_or("valve");
+        let is_paranoia = game_dir == "paranoia";
+        let is_cstrike = game_dir == "cstrike";
+        let is_czero = game_dir == "czero";
+
+        let max_speed = get_cvar_f32(marker, "sv_maxspeed").unwrap_or(320.);
+        let client_max_speed = engine::pmove
+            .get_opt(marker)
+            .map(|pmove| (**pmove).clientmaxspeed)
```

Comment:
I see BXT uses `GetClientMaxspeed()` for this. I'm not sure using `pmove` is equivalent, isn't it one frame off? Since `pmove` is only filled/updated later in the frame.

Reply:
I am not aware of it being one frame off. It does sync well on my end. If it is one frame off then it would be one frame off the very first frame of the TAS as weapon speed would be the same the whole run.

Reply:
Also, it is in the bind, so I just use whatever is there.